src/main/java/org/plumelib/bcelutil/SimpleLog.java

```diff
@@ -0,0 +1,98 @@
+package org.plumelib.bcelutil;
+
+/*>>>
+import org.checkerframework.checker.formatter.qual.*;
+import org.checkerframework.checker.initialization.qual.*;
+import org.checkerframework.checker.nullness.qual.*;
+*/
+
+/**
+ * A logging class with the following features:
+ *
+ * <ul>
+ *   <li>Can be enabled and disabled (when disabled, all operations are no-ops),
+ *   <li>Can indent/exdent log output,
+ *   <li>Writes to standard output, and
+ *   <li>Can provide a backtrace
+ * </ul>
+ */
+public final class SimpleLog {
+
+  /** If false, do no output. */
+  public boolean enabled;
+
+  /** The current indentation string. */
+  private String indent_str = "";
+  /** Indentation string for one level of indentation. */
+  public final String INDENT_STR_ONE_LEVEL = "  ";
+
+  /**
+   * Create a new SimpleLog object.
+   *
+   * @param enabled whether the logger starts out enabled
+   */
+  public SimpleLog(boolean enabled) {
+    this.enabled = enabled;
+  }
+
+  public SimpleLog() {
+    this(true);
+  }
+
+  public boolean enabled() {
+    return enabled;
+  }
+
+  /**
+   * Log a message.
+   *
+   * @param format format string for message
+   * @param args values to be substituted into format
+   */
+  /*@FormatMethod*/
+  public void log(String format, /*@Nullable*/ Object... args) {
+    if (enabled) {
+      System.out.print(indent_str);
+      System.out.printf(format, args);
```

Comment:
Should `indent_str` be inserted at every line break in the formatted output?
More generally, the semantics of indentation are not defined.